src/kibana/components/vislib/visualizations/tile_map.js

```diff
@@ -160,29 +249,120 @@ define(function (require) {
     };
 
     /**
+     * Finds nearest feature in mapData to event latlng
+     *
+     * @method nearestFeature
+     * @param point {Leaflet Object}
+     * @param mapData {geoJson Object}
+     * @return nearestPoint {Leaflet Object}
+     */
+    TileMap.prototype.nearestFeature = function (point, mapData) {
+      var self = this;
+      var distance = Infinity;
+      var nearest;
+
+      if (point.lng < -180 || point.lng > 180) {
+        return;
+      }
+
+      for (var i = 0; i < mapData.features.length; i++) {
+        var dist = point.distanceTo(mapData.features[i].properties.latLng);
+        if (dist < distance) {
+          distance = dist;
+          nearest = mapData.features[i];
+        }
+      }
+      nearest.properties.eventDistance = distance;
+
+      return nearest;
+    };
+
+    /**
+     * display tooltip if feature is close enough to event latlng
+     *
+     * @method tooltipProximity
+     * @param latlng {Leaflet Object}
+     * @param zoom {Number}
+     * @param feature {geoJson Object}
+     * @param map {Leaflet Object}
+     * @return boolean
+     */
+    TileMap.prototype.tooltipProximity = function (latlng, zoom, feature, map) {
+      if (!feature) {
+        return;
+      }
+
+      var showTip = false;
+      var zoomScale = d3.scale.linear()
+      .domain([1, 9, 14, 18])
+      .range([750000, 25000, 10, 0.01]);
+
+      var lngDif = Math.abs(latlng.lng - feature.properties.latLng.lng);
+      var proximity = zoomScale(zoom);
+      var distance = latlng.distanceTo(feature.properties.latLng);
+
+      if (distance < proximity && lngDif < 40) {
```

Comment:
Again, what is the 40 for? See here: https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#magic-numbers